### PR TITLE
Finance react app 

### PR DESCRIPTION
**Pull request by ridabensalem**
In this milestone of reacting capstone, we have added the following requirements : 

- Choosed  the finance API 
- Fetched the endpoints of companies, sectors, and profile
- create the redux logic for stock, companies, and companies details 
- When the user clicks a stock it displays a list of companies ]
- Implement a search bar to filter the list of companies 
- when the user clicks the backward icon the app returns to the previous page 
- show details of the companies 
- Added routes to switch  between pages 
- Added tests for the app 
- Styled the app following the guidelines of Behance 
- Deployed the app

### CODE REVIEW

File: src/testing/apitest.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  test('fetchSectors should return an array of sectors', async () => {
+    await fetchSectors()(dispatch);
+    expect(dispatchedActions).toHaveLength(2);
+    expect(dispatchedActions[0].type).toEqual('sectors/fetchSectors/pending');
+    expect(dispatchedActions[1].type).toEqual('sectors/fetchSectors/fulfilled');
+    expect(dispatchedActions[1].payload).toBeDefined();
+    expect(Array.isArray(dispatchedActions[1].payload)).toBe(true);
+  });
+});
```
Comment (bobb-Rob) [inline src/testing/apitest.test.js:31]: - [ ] I see that you wrote tests for Async Thunk API Functions. Well done. But you are expected to write **_UNIT_** tests for several components of the app and some pure functions in Redux code. Also, your App should have integration tests using React Testing Library

* Student can perform a mocking data fetching and Redux Store connection.

Kindly add more tests for your components, you can even add screenshots test.

**See project requirement below 👇🏾** 
image

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @ridabensalem, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️

Wow, well done on this project. 
- [ ] Based on the project requirement, you are suppose to add a a vdoe instead you added a `gif`  file withoutany voice. The video should be a maximum length of 5mins and should include the following:  
     - The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
     - The video contains a demo of the project, highlighting all the project features.
     - You should also highlight some interesting piece of code or something you built that you are very proud of.

For recording the video you can use tools like [Loom](https://www.loom.com/) that let you share a private link to the recording, and configure a shot that shows your computer screen and your face at the same time in a small picture.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @ridabensalem,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShadyShawkat — APPROVED** [review]
## Hi @ridabensalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**